src/modules/Game.class.js

```diff
@@ -6,63 +6,293 @@
  * Feel free to add more props and methods if needed.
  */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  static gameStatus = {
+    idle: 'idle',
+    playing: 'playing',
+    win: 'win',
+    lose: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.status = Game.gameStatus.idle;
+    this.score = 0;
+    this.initialState = initialState;
+    this.state = initialState.map((row) => [...row]);
+  }
+
+  moveLeft() {
+    if (this.getStatus() !== Game.gameStatus.playing) {
+      return;
+    }
+
+    let hasMoved = false;
+
+    for (let i = 0; i < this.state.length; i++) {
+      let filteredRow = this.state[i].filter((value) => value !== 0);
+
+      for (let j = 0; j < filteredRow.length - 1; j++) {
+        if (filteredRow[j] === filteredRow[j + 1]) {
+          filteredRow[j] *= 2;
+          filteredRow[j + 1] = 0;
+          this.score += filteredRow[j];
+          hasMoved = true;
+        }
+      }
+
+      filteredRow = filteredRow.filter((value) => value !== 0);
+
+      while (filteredRow.length < this.state[i].length) {
+        filteredRow.push(0);
+      }
+
+      if (this.state[i].toString() !== filteredRow.toString()) {
+        hasMoved = true;
+        this.state[i] = filteredRow;
+      }
```

Comment:
The logic within the 'moveLeft' method is correct, but it's not immediately clear what it does. Consider adding comments explaining the purpose of each step, or refactor the code into smaller functions with descriptive names to improve readability.